modules/nf-core/gsea/gsea/main.nf

```diff
@@ -26,29 +26,27 @@ process GSEA_GSEA {
     tuple val(meta), path("*pvalues_vs_nes_plot.png")          , emit: pvalues_vs_nes_plot
     tuple val(meta), path("*ranked_list_corr_2.png")           , emit: ranked_list_corr
-    tuple val(meta), path("*butterfly_plot.png")               , emit: butterfly_plot, optional: true
-    tuple val(meta), path("*[!gene_set_size|gsea_report|ranked_gene_list]*.tsv"), emit: gene_set_tsv, optional: true
-    tuple val(meta), path("*[!gsea_report|heat_map_corr_plot|index|pos_snapshot|neg_snapshot]*.html"), emit: gene_set_html, optional: true
-    tuple val(meta), path("*[!butterfly|enplot|global_es_histogram|gset_rnd_es_dist|heat_map|pvalues_vs_nes_plot|ranked_list_corr]*.png"), emit: gene_set_heatmap, optional: true
-    tuple val(meta), path("*_snapshot*.html")                  , emit: snapshot, optional: true
-    tuple val(meta), path("*enplot*.png")                      , emit: gene_set_enplot, optional: true
-    tuple val(meta), path("*gset_rnd_es_dist*.png")            , emit: gene_set_dist, optional: true
-    tuple val(meta), path("*.zip")                             , emit: archive, optional: true
+    tuple val(meta), path("*butterfly_plot.png")               , emit: butterfly_plot  , optional: true
+    tuple val(meta), path("gene_sets_*.tsv")                   , emit: gene_set_tsv    , optional: true
+    tuple val(meta), path("gene_sets_*.html")                  , emit: gene_set_html   , optional: true
+    tuple val(meta), path("gene_sets_*.png")                   , emit: gene_set_heatmap, optional: true
+    tuple val(meta), path("*_snapshot*.html")                  , emit: snapshot        , optional: true
+    tuple val(meta), path("*enplot*.png")                      , emit: gene_set_enplot , optional: true
+    tuple val(meta), path("*gset_rnd_es_dist*.png")            , emit: gene_set_dist   , optional: true
+    tuple val(meta), path("*.zip")                             , emit: archive         , optional: true
     path "versions.yml"                                        , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def VERSION = '4.3.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-
-    // Remove any trailing dots from prefix when passed as report label, so
-    // GSEA doesn't produce double-dotted top-level outputs
-    def rpt_label = prefix.replaceAll('\\.$', '')
-
+    def rpt_label = prefix.replaceAll('\\.$', '') // Remove any trailing dots from prefix when passed as report label, so GSEA doesn't produce double-dotted top-level outputs
     def chip_command = chip ? "-chip $chip -collapse true" : ''
+    def VERSION = '4.3.2' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    if (!(args ==~ /.*-rnd_seed.*/)) {args += " -rnd_seed 10"}
```

Comment:
We should not change the default behaviour of underlying software unless we have to.

Set this for tests, but not here.